benches/oneshot/ipa.rs

```diff
@@ -1,62 +1,57 @@
-use rand::Rng;
+use rand::{rngs::StdRng, thread_rng, Rng, SeedableRng};
 use raw_ipa::{
     error::Error,
-    ff::Fp32BitPrime,
-    ipa_test_input,
-    protocol::{ipa::ipa, BreakdownKey, MatchKey},
-    test_fixture::{input::GenericReportTestInput, Runner, TestWorld, TestWorldConfig},
+    test_fixture::{
+        generate_random_user_records_in_reverse_chronological_order, test_ipa,
+        update_expected_output_for_user, IpaSecurityModel, TestWorld,
+    },
 };
-use std::{num::NonZeroUsize, time::Instant};
 
 #[tokio::main(flavor = "multi_thread", worker_threads = 3)]
 async fn main() -> Result<(), Error> {
-    const BATCHSIZE: usize = 100;
-    const MAX_TRIGGER_VALUE: u128 = 5;
-    const PER_USER_CAP: u32 = 3;
-    const MAX_BREAKDOWN_KEY: u128 = 4;
-    const NUM_MULTI_BITS: u32 = 3;
-
-    let mut config = TestWorldConfig::default();
-    config.gateway_config.send_buffer_config.items_in_batch = NonZeroUsize::new(1).unwrap();
-    config.gateway_config.send_buffer_config.batch_count = NonZeroUsize::new(1024).unwrap();
-    let world = TestWorld::new_with(config).await;
-    let mut rng = rand::thread_rng();
-
-    let max_match_key = u128::try_from(BATCHSIZE / 4).unwrap();
-
-    let mut records: Vec<GenericReportTestInput<Fp32BitPrime, MatchKey, BreakdownKey>> =
-        Vec::with_capacity(BATCHSIZE);
-
-    for _ in 0..BATCHSIZE {
-        records.push(ipa_test_input!(
-            {
-                match_key: rng.gen_range(0..max_match_key),
-                is_trigger_report: rng.gen::<u32>(),
-                breakdown_key: rng.gen_range(0..MAX_BREAKDOWN_KEY),
-                trigger_value: rng.gen_range(0..MAX_TRIGGER_VALUE),
-            };
-            (Fp32BitPrime, MatchKey, BreakdownKey)
-        ));
+    const MAX_BREAKDOWN_KEY: usize = 16;
+    const MAX_TRIGGER_VALUE: u32 = 5;
+    const NUM_USERS: usize = 8;
+    const MAX_RECORDS_PER_USER: usize = 10;
```

Comment:
The randomness here will make it more difficult to benchmark IPA on N records. The total number of records generated will now be a probabilistic outcome. That's OK, but we might want to consider the usecase of "How fast is N rows?" and make some way to do that.